### PR TITLE
[chassisd] Add script to initialize chassis info in STATE_DB

### PR DESCRIPTION
**Pull request by robocoder99**
#### Description
I added a new script `chassis_db_init` that uploads chassis hardware information such as serial number, model number and hardware revision to the STATE_DB under the CHASSIS_INFO table. 

#### Motivation and Context
I made this change in order to expose the chassis hardware information to SONiC user space and allow CLI utilities to access it in order to expose it to the user. 

#### How Has This Been Tested?
Unit tests were added and was deployed and tested on Mellanox platform MSN2700.

### CODE REVIEW

File: sonic-chassisd/tests/test_chassis_db_init.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    revision = "Rev C"
+
+    chassis_table = provision_db(chassis, log)
+
+    fvs = chassis_table.get(CHASSIS_INFO_KEY_TEMPLATE.format(1))
+    assert serial == fvs[CHASSIS_INFO_SERIAL_FIELD]
+    assert model == fvs[CHASSIS_INFO_MODEL_FIELD]
+    assert revision == fvs[CHASSIS_INFO_REV_FIELD]
+
+
```
Comment (jleveque) [inline sonic-chassisd/tests/test_chassis_db_init.py:40]: Delete trailing blank lines

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by jleveque** [comment]
Also please update PR description, as it references `chassisshow`, which is not part of this PR.

**Comment by jleveque** [comment]
Just curious why a separate script is necessary for this. Why can we not simply add this logic to chassisd when it starts up?

**Comment by robocoder99** [comment]
We decided on a separate script because chassisd is primarily for handling modular chassis management (and thus is only run on modular chassis platforms), a separate task from uploading basic information such as serial, model, and hardware revision that is applicable to all platforms and also one-shot.

**Review by jleveque — APPROVED** [review]

**Comment by jleveque** [comment]
Since this script is a one-shot script, and not a daemon per se, I have reservations about adding it in this repo. We could simply add it in the buildimage repo as part of https://github.com/Azure/sonic-buildimage/pull/7596. However, it would not be built into a package, so we would not be able to run unit tests against it. I'm on the fence, but I guess I can live with having it as part of the chassisd package.

**Comment by robocoder99** [comment]
I would argue that while the name `sonic-platform-daemons` does indicate a one-shot task wouldn't belong, architecturally it does belong here as it performs the same fundamental task as the chassisd daemon (uploading chassis information to the redis db). As such anyone looking for this code would likely expect to find it here and I think that is valuable. 

If there exists now or in the future a collection of one-shot tasks for the purpose of provisioning redis then I would argue this should be refactored at that point.

**Comment by jleveque** [comment]
> architecturally it does belong here as it performs the same fundamental task as the chassisd daemon (uploading chassis information to the redis db).

I agree with this, which is why I am on the fence.

> If there exists now or in the future a collection of one-shot tasks for the purpose of provisioning redis then I would argue this should be refactored at that point.

Agreed. We can proceed with this PR.

**Pull request merged**